src/c++/perf_analyzer/mpi_utils.h

```diff
@@ -0,0 +1,84 @@
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <memory>
+
+namespace triton { namespace perfanalyzer {
+
+class MPIDriver {
+ public:
+  // Initializes class. Saves handle to MPI library if MPI library is available.
+  MPIDriver(bool verbose = false);
+
+  // Returns true if the current process is an MPI process with world size
+  // greater than 1.
+  bool IsMPIRun();
+
+  // Attempts to call MPI_Init API.
+  void MPIInit(int* argc, char*** argv);
+
+  // Attempts to call MPI_Comm_size API with MPI_COMM_WORLD communicator.
+  int MPICommSizeWorld();
+
+  // Attempts to call MPI_Barrier API with MPI_COMM_WORLD communicator.
+  void MPIBarrierWorld();
+
+  // Attempts to call MPI_Comm_rank API with MPI_COMM_WORLD communicator.
+  int MPICommRankWorld();
+
+  // Attempts to call MPI_Bcast API with MPI_INT data type and MPI_COMM_WORLD
+  // communicator.
+  void MPIBcastIntWorld(void* buffer, int count, int root);
+
+  // Attempts to call MPI_Finalize API.
+  void MPIFinalize();
+
+ private:
+  // Attempts to call MPI_Initialized API.
+  bool MPIInitialized();
+
+  // Returns MPI_COMM_WORLD symbol address if MPI library is available,
+  // otherwise `nullptr`.
+  void* MPICommWorld();
+
+  // Returns MPI_INT symbol address if MPI library is available, otherwise
+  // `nullptr`.
+  void* MPIInt();
+
+  // Attempts to check that Open MPI is installed.
+  void CheckMPIImpl();
+
+  // Loaded object for MPI library.
+  void* handle_{nullptr};
+
+  // Maximum string length for MPI version string.
+  const uint64_t MPIVersionStringMaximumLength{32768};
+};
+
+using MPIDriverPtr_t = std::shared_ptr<MPIDriver>;
```

Comment:
Why is this needed? Wouldn't explicitly calling `std::shared_ptr<MPIDriver>` everywhere instead of MPIDriverPtr_t would make ownership logic more clear?

Reply:
Removed type alias 👍